llm/llm.go

```diff
@@ -248,19 +248,27 @@ func (l *LLMImpl) attemptGenerate(ctx context.Context, prompt *Prompt) (string,
 
 	// Extract and log caching information
 	var fullResponse map[string]interface{}
-	if err := json.Unmarshal(body, &fullResponse); err == nil {
-		if usage, ok := fullResponse["usage"].(map[string]interface{}); ok {
-			l.logger.Debug("Usage information", "usage", usage)
-			cacheInfo := map[string]interface{}{
-				"cache_creation_input_tokens": usage["cache_creation_input_tokens"],
-				"cache_read_input_tokens":     usage["cache_read_input_tokens"],
+	if err := json.Unmarshal(body, &fullResponse); err != nil {
```

Comment:
Here's the meat of the PR. If JSON parsing fails, we try parsing JSONL and getting the last entry.